### PR TITLE
DAShboard status page changing both camera video feed and wireless modules online/offline using status

### PR DESCRIPTION
**Pull request by ericyem**
## Description

Changed the the way wireless modules are turned online, should be done by sending the payload through the wireless_module/#/status topic. Fixed the video feed so that it displays "On" or "Off" depending on the camera status feed.
## Screenshots

image

## Steps to Test

1. Run the DAS server
2. To test the wireless modules going online, send this message: `mosquitto_pub -t "/v3/wireless_module/2/status" -h localhost -m "{\"online\": true}"` and if you want to turn it offline send the same message with false instead of true.
3. To test the camera overlay, take raspicam repo and run: `python orchestrator.py` and `python overlay_new.py --host localhost (need a camera or if background run with --bg (name of background image))`. This should display online in the camera status for the status overview, and the feed should also turn online. Can also test by sending `"status/camera/video_feed/primary" -h localhost -m "{\"online\": true}"` after turning on the camera using `mosquitto_pub -t "status/camera/primary" -h localhost -m "{\"connected\": true}"`

### CODE REVIEW

File: server/js/sockets.js
Comment (kdav108) [inline server/js/sockets.js]: If this is only for testing, we can remove it so it doesn't go into `master`

File: server/js/sockets.js
Comment (kdav108) [inline server/js/sockets.js]: We can get rid of this line

File: server/js/sockets.js
Comment (kdav108) [inline server/js/sockets.js]: Two nits:
1) Can we move both of these to where the other subscription takes place (i.e. `271`) just so they're grouped together and easy to find
2) Can we use `'${Camera.base}/#'` instead of 'hardcoding' the stem/prefix of the topic

**Review by kdav108 — CHANGES_REQUESTED** [review]
Nice work, great to see that page finally working correctly. 

Just got a few small requests...

**Comment by kdav108** [comment]
This is a good PR to review @bahar-ap since it's related to the page you're working on. I'll let you do the honour of approving this PR once you've reviewed and tested it :)

File: server/js/sockets.js
Comment (ericyem) [reply server/js/sockets.js]: Yup forgot about that removed!

File: server/js/sockets.js
Comment (ericyem) [reply server/js/sockets.js]: Done!

File: server/js/sockets.js
Comment (ericyem) [reply server/js/sockets.js]: Done!

**Review by kdav108 — COMMENTED** [review]
Looks good to me! Waiting on @bahar-ap's view :)

**Review by hallgchris — CHANGES_REQUESTED** [review]
Not sure if I'm totally missing something or if something on my computer is set up wrong... But this PR doesn't seem to be working at all for me.

Publishing `mosquitto_pub -t "status/camera/video_feed/primary" -h localhost -m "{\"online\": true}"` doesn't make the camera appear online.

`mosquitto_pub -t "/v3/wireless_module/2/status" -m "{\"online\": true}"` works, however when I refresh the page it isn't retained.

Is the latest commit definitely functioning for you? I haven't dived into the code yet, so if it works for you it's probably my end and I'll take a look.

Aside from that, can you give this PR a descriptive name? Also, there are some conflicts in `sockets.js` you'll need to fix before merging.

**Comment by ericyem** [comment]
> Not sure if I'm totally missing something or if something on my computer is set up wrong... But this PR doesn't seem to be working at all for me.
> Publishing `mosquitto_pub -t "status/camera/video_feed/primary" -h localhost -m "{\"online\": true}"` doesn't make the camera appear online.

I think I know the reason why it might be because the camera is switched on yet? I can change it so that camera turns on if the video feed switches on if that should be intended.

> `mosquitto_pub -t "/v3/wireless_module/2/status" -m "{\"online\": true}"` works, however when I refresh the page it isn't retained.

Yup I'll look into that 

> Aside from that, can you give this PR a descriptive name? Also, there are some conflicts in `sockets.js` you'll need to fix before merging.

Done!  Updated PR title

File: server/js/sockets.js
Comment (haotongwang) [inline server/js/sockets.js:294]: What is the "for wireless module" comment here describing?

File: server/js/sockets.js
Comment (haotongwang) [inline server/js/sockets.js:281]: Should there be a subscription to the wireless module status topic here?

File: server/js/sockets.js
Comment (ericyem) [reply server/js/sockets.js:294]: Oh it was just for me sorry because I was about to make a function so that the retained data in the server were kept on reload, but then I realised that that function basically does that so I can remove that.

File: server/js/sockets.js
Comment (ericyem) [reply server/js/sockets.js:281]: image
This part of the code should be the one subscribing to the wireless module status topic if I'm not mistaken so should be working as intended?

Comment (haotongwang) [reply server/js/sockets.js:281]: Yes, you're right, forgot about that line.

**Review by kdav108 — APPROVED** [review]
LGTM!

Just a suggestion, unrelated to the changes brought in with this PR, but since it deals with the same UI component, how about removing that `501 Not Implemented` message:

Screen Shot 2021-09-25 at 6 58 46 pm

**Comment by ericyem** [comment]
image
For the 501 Not Implemented there's a TODO for "use api". does anyone know what means before I delete it?

File: server/js/sockets.js
Comment (hallgchris) [inline server/js/sockets.js:28]: This comment can go!

**Review by hallgchris — APPROVED** [review]
LGTM!

About the 501 error. I think Kunj is suggesting you could implement the front end for displaying the display's IP address? It wouldn't be a big change, up to you if you do it here.

Basically, we now publish the display's IP address in the status message (as documented [here](https://www.notion.so/monashhumanpower/V3-MQTT-Topics-66e6715d0e1941ffaa82020e5868fbae#9678c43a06cc48d6885d825a0344ee58)). You would need to update the `CameraStatus` type (`camera.ts`) and then use the new value on the lines where we currently have `501 not implemented`.

File: server/js/sockets.js
Comment (ericyem) [reply server/js/sockets.js:28]: Yeah I'll just do it in a nother pull request then just want this to be merged first if thats okay haha

Comment (hallgchris) [reply server/js/sockets.js:28]: No worries! In the future, if you get an approval with a really small suggestion you can just make the change and then merge straight away :)

**Pull request merged**
